### PR TITLE
mysql-client mysql 8.1.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
> Important Change: Dropped support for Enterprise Linux 6 (and associated glibc 2.12 generic), SUSE 12, Debian 10, MacOS 12, Ubuntu 18.04 and 20.04, Windows 10 and Server 2012R2; and 32-bit versions are no longer built.

**Comment by p-linnane** [comment]
v8.0 is supported through April 2026. We should create a versioned formula for it in this PR.

**Comment by SMillerDev** [comment]
I think we should leave the versioned formula unless it's really needed. We're talking about the client here, not the server component. Not a lot changes in this.

**Comment by chenrui333** [comment]
yeah, we can followup on mysql 8.0 stuff (nb it is also a LTS release, see blog, https://blogs.oracle.com/mysql/post/introducing-mysql-innovation-and-longterm-support-lts-versions)

File: Formula/mysql-client.rb [hunk 1]
```diff
@@ -29,17 +28,13 @@ class MysqlClient < Formula
   # GCC is not supported either, so exclude for El Capitan.
   depends_on macos: :sierra if DevelopmentTools.clang_build_version < 900
   depends_on "openssl@3"
-  depends_on "zlib" # Zlib 1.2.12+
```
Comment (carlocab) [inline Formula/mysql-client.rb:32]: Why remove this?

File: Formula/mysql.rb [hunk 2]
```diff
@@ -2,11 +2,10 @@ class Mysql < Formula
   desc "Open source relational database management system"
   homepage "https://dev.mysql.com/doc/refman/8.0/en/"
   # TODO: Check if we can use unversioned `protobuf` at version bump
-  # https://bugs.mysql.com/bug.php?id=111469
```
Comment (carlocab) [inline Formula/mysql.rb:5]: I don't think this is fixed yet in 8.1.

File: Formula/mysql.rb
Comment (carlocab) [inline Formula/mysql.rb:1]: Needs `depends_on "bison" => :build`

Comment (osalbahr) [reply Formula/mysql.rb:1]: Hi @chenrui333, did adding bison as a build dependency resolve the error?

File: Formula/mysql-client.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/mysql-client.rb:32]: somehow I thought it was resolved.

File: Formula/mysql.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/mysql.rb:5]: 👍

**Comment by roshanjrajan-zip** [comment]
hi @chenrui333 is there anything I can do to help to get this through?

**Comment by ZhongRuoyu** [comment]
> pushed some updates, but still need some help on this build error

Build looks fine?

[Failures](https://github.com/Homebrew/homebrew-core/actions/runs/5879638902/job/15947508293?pr=136966) are all with linkage:

```
Error: 23 failed steps!
brew linkage --test diesel
brew test --retry --verbose diesel
brew linkage --test hydra
brew test --retry --verbose hydra
brew linkage --test innotop
brew linkage --test inspircd
brew linkage --test kitchen-sync
brew test --retry --verbose kitchen-sync
brew linkage --test libzdb
brew test --retry --verbose libzdb
brew linkage --test mydumper
brew test --retry --verbose mydumper
brew linkage --test mysql++
brew test --retry --verbose mysql++
brew linkage --test ntopng
brew test --retry --verbose ntopng
brew linkage --test percona-toolkit
brew linkage --test root
brew linkage --test sysbench
brew test --retry --verbose sysbench
brew linkage --test percona-xtrabackup
brew linkage --test qt-mysql
brew test --retry --verbose qt-mysql
```

**Comment by chenrui333** [comment]
We can followup mysql@8.0 after this PR gets merged

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**